### ISSUE TITLE
Webpack 5: lintDirtyModulesOnly fails because getDirtyFiles returns lots of non-existent files

### ISSUE THREAD

**Issue by jrnail23** [webpack-contrib/eslint-webpack-plugin#32]
- Operating System: Mac
- Node Version: 10.19.0
- NPM Version:  6.13.4
- webpack Version: 5.2.0
- eslint-webpack-plugin Version: 2.1.0

### Expected Behavior / Situation
When using `lintDirtyModulesOnly` w/Webpack 5, only real, dirty files should be linted.

### Actual Behavior / Situation
For reference sake, here's my configuration:
```js
new ESLintWebpackPlugin({
  files: ['src/client', 'src/shared'],
  extensions: ['js', 'jsx'],
  cache: true,
  lintDirtyModulesOnly: true,
}),
```

Upon modifying a watched file, I get an error like this: `ERROR in No files matching '/Users/me/my-project/src/client/Component.js' were found`

Via a debugger, I found that for some reason `runCompiler.fileTimestamps` contains lots and lots of non-existent files.  I think that's a webpack issue, perhaps due to my usage of both `.js` and `.jsx` file extensions in my project (or who knows why), but it bleeds into `eslint-webpack-plugin` by causing `_DirtyFileWatcher.getDirtyFiles` to return pretty much all of the non-existent file paths that fall under my context (not just dirty, real ones).

To be clear, these non-existent files usually look something like this in the fileTimestamps map: `{'my-project-path/some-actual-js-file.jsx' => null}` (where the real `actual-js-file.jsx` actually has a `.js` file extension).

On one hand, this bit of logic in the DirtyFileWatcher is a real head-scratcher:
```js
const hasFileChanged = (filename, fileSystemInfoEntry) => {
  const prevTimestamp = getTimestamps(this.prevTimestamps.get(filename));
  const timestamp = getTimestamps(fileSystemInfoEntry);

  // this comparison logic will always return any file that has a null timestamp, 
  // even though a null timestamp seems to mean the file doesn't exist!
  return (prevTimestamp || this.startTime) < (timestamp || Infinity);
};
```

When I changed that code locally to the following, it just worked:
```js
const hasFileChanged = (filename, fileSystemInfoEntry) => {
  const prevTimestamp = getTimestamps(this.prevTimestamps.get(filename));
  const timestamp = getTimestamps(fileSystemInfoEntry);
  if (!timestamp) {
    return false;
  }
  return (prevTimestamp || this.startTime) < (timestamp);
};
```

Using Webpack5's newly added `runCompiler.modifiedFiles` property makes this problem much easier to solve.  I simply wrapped the existing plugin with the following, so I could substitute my own dirty modules logic (this way I have a temporary workaround that doesn't involve me forking this project):

```js
const ESLintWebpackPlugin = require('eslint-webpack-plugin');
const linter = require('eslint-webpack-plugin/dist/linter').default;

module.exports = class Webpack5ESLintWebpackPlugin extends ESLintWebpackPlugin {
  constructor(options = {}) {
    super(options);
  }

  apply(compiler) {
    const {lintDirtyModulesOnly, ...options} = this.options;
    if (!lintDirtyModulesOnly) {
      super.apply(compiler);
      return;
    }

    compiler.hooks.watchRun.tapPromise(
      'ESLintWebpackPlugin',
      async (runCompiler) => {
        const {modifiedFiles} = runCompiler;
        if (modifiedFiles && modifiedFiles.size > 0) {
          await linter(
            {
              ...options,
              files: Array.from(modifiedFiles.values()),
            },
            runCompiler
          );
        }
      }
    );
  }
};

```

### Modification Proposal
Basically any of the above hacks should resolve the issue (at least naively), but I figured I'd share what I did to help illustrate the problem. Obviously my `modifiedFiles` hack doesn't address filtering by files or extensions, but it's helpful to demonstrate the issue (I'll definitely need to deal with extensions before using it in anger). 

I'd recommend checking for the existence of the `modifiedFiles` property, and if it exists, skip the dirtyFileWatcher entirely.
If it doesn't exist, that implies one of the following:
- either we're using an earlier version of webpack (can confirm via `runCompiler.webpack.version`), or
- It's the first run (no files have been modified yet).

**Comment by undermuz** [comment]
I have same problem on windows with webpack@5.3.2

**Comment by ricardogobbosouza** [comment]
Hi @jrnail23 
Try [v2.2.0](https://github.com/webpack-contrib/eslint-webpack-plugin/releases/tag/v2.2.0)

**Comment by jrnail23** [comment]
@ricardogobbosouza trying it now. Thanks for the heads up. I'll let you know what happens.￼

**Comment by jrnail23** [comment]
@ricardogobbosouza, I just tried v2.2.0, and it doesn't seem to be raising eslint warnings at all when using `lintDirtyModulesOnly` in `watch` mode.

I'm using the same config as above:
```js
new ESLintWebpackPlugin({
  files: ['src/client', 'src/shared'],
  extensions: ['js', 'jsx'],
  cache: true,
  lintDirtyModulesOnly: true,
}),
```

**Comment by ricardogobbosouza** [comment]
😄

**Comment by jrnail23** [comment]
@ricardogobbosouza, to be clear, that's a bad thing.  It's not outputting eslint errors or warnings when I add them to a file in watch mode.

**Comment by ricardogobbosouza** [comment]
@jrnail23 
If you can share the repository it would be of great help

**Comment by ricardogobbosouza** [comment]
@jrnail23 Remembering that only files imported on the graph will be linted, at least it is the expected behavior.

Ex:
**_src/index.js_**
```js
import('./lint')
```

**_src/lint.js_**
```js
// this file will linted
const foo = 'bar'
```

**_src/no-lint.js_**
```js
// this file will not linted, why not being used
```

**_webpack.config.js_**
```js
const ESLintPlugin = require('eslint-webpack-plugin');

module.exports = {
  plugins: [new ESLintPlugin({lintDirtyModulesOnly: true})],
};
```

**Comment by jrnail23** [comment]
@ricardogobbosouza, it's the globs.  Since the my `files` option didn't contain glob paths, `processModule` isn't matching them.  That's a breaking change, I think.
Also, in the previous version, running watch with `lintDirtyModulesOnly` would only run the linter after files were changed.  Now it lints the whole graph on the first run -- that makes the initial build MUCH slower, killing much of `lintDirtyModulesOnly`'s value.

**Comment by ricardogobbosouza** [comment]
Hi @jrnail23 
Resolved v2.3.0
If the problem persists, reopen this
